tests/test_vision_data.py

```diff
@@ -165,7 +165,7 @@ def test_download_images():
 def test_trunc_download():
     this_tests(untar_data)
     url = URLs.COCO_TINY
-    fname = datapath4file(f'{url2name(url)}.tgz')
+    fname = datapath4file(f'{url2name(url)}.tgz', archive=True)
```

Comment:
If this feature didn't exist, how can it be that the test that didn't not require it, now requires it?

Reply:
So in the old implementation, `datapath4file` is used to both get the path for decompressed datasets and the compressed dataset file, since they are strictly in the same path if the user does not specify `fname` in `download_data` or `dest` in `untar_data`. This will cause an issue if we separate archived files and decompressed files in two different paths, so we need an API change. 

In this particular case, the test is looking for the archive path to compare checksum, so I need to add archive=True to the end.